clients/vscode/src/chat/ChatViewProvider.ts

```diff
@@ -10,18 +11,25 @@ export class ChatViewProvider implements WebviewViewProvider {
   client?: ServerApi;
   private pendingMessages: ChatMessage[] = [];
 
-  constructor(private readonly context: ExtensionContext) {}
+  constructor(
+    private readonly context: ExtensionContext,
+    private readonly config: Config,
+  ) {}
 
   public async resolveWebviewView(webviewView: WebviewView) {
     this.webview = webviewView;
     const extensionUri = this.context.extensionUri;
-    const { server } = agent().getConfig();
 
     webviewView.webview.options = {
       enableScripts: true,
       localResourceRoots: [extensionUri],
     };
-    webviewView.webview.html = await this._getWebviewContent();
+    console.log("resolveWebviewView", { config: this.config.server });
+    webviewView.webview.html = this._getWebviewContent(this.config.server);
+    this.config.on("updatedServerConfig", () => {
+      console.log("updatedServerConfig", { config: this.config.server });
+      webviewView.webview.html = this._getWebviewContent(this.config.server);
+    });
```

Comment:
@wwayne  I added a listener for changes ofserver config.

Reply:
not sure if will affect other stuff like theme syncing, I will do more testing after merge